examples/exampleFMU/exampleFMU2_MSRE_HX.f90

```diff
@@ -0,0 +1,167 @@
+
+  WRITE(*,*) "Is T_in valid: ", test_fmu2_slave%isXmlVar(T_in_name)
+  WRITE(*,*) "Is P_in valid: ", test_fmu2_slave%isXmlVar(P_in_name)
+  WRITE(*,*) "Is dummy var valid: ", test_fmu2_slave%isXmlVar(junk_var_name)
+
+  ! Initialization loop, let system come to Steady State
+  DO i=1,2000
+    CALL test_fmu2_slave%doStep(1.0E-2_SRK)
+  ENDDO
+
+  ! set restart point
+  CALL test_fmu2_slave%setRestart()
+  WRITE(*,*) "Set restart"
+
+  WRITE(*,*) "time(s), T_in(K), T_out(K), mflow_out(kg/s), P_out(Pa)"
+  DO i=1,3000
+    CALL test_fmu2_slave%getNamedVariable(T_out_name, T_out)
+    CALL test_fmu2_slave%getNamedVariable(mflow_out_name, mflow_out)
+    CALL test_fmu2_slave%getNamedVariable(mflow_pump_name, mflow_pump)
+    CALL test_fmu2_slave%getNamedVariable(P_out_name, P_out)
+    CALL test_fmu2_slave%getNamedVariable(T_in_name, T_in_out)
+    CALL test_fmu2_slave%doStep(dt)
+    time=time+dt
+    WRITE(*,*) time, T_in_out, T_out, mflow_out, mflow_pump, P_out
+  ENDDO
+
+  ! increase inlet T
+  CALL test_fmu2_slave%setNamedVariable(T_in_name, 927.0_SRK)
+
+  DO i=1,3000
+    CALL test_fmu2_slave%getNamedVariable(T_out_name, T_out)
+    CALL test_fmu2_slave%getNamedVariable(mflow_out_name, mflow_out)
+    CALL test_fmu2_slave%getNamedVariable(mflow_pump_name, mflow_pump)
+    CALL test_fmu2_slave%getNamedVariable(P_out_name, P_out)
+    CALL test_fmu2_slave%getNamedVariable(T_in_name, T_in_out)
+    CALL test_fmu2_slave%doStep(dt)
+    time=time+dt
+    WRITE(*,*) time, T_in_out, T_out, mflow_out, mflow_pump, P_out
+  ENDDO
+
+  ! Decrease the flow rate, keep T_in the same
+  CALL test_fmu2_slave%setNamedVariable(mflow_in_name, 160.0_SRK)
+  CALL test_fmu2_slave%setNamedVariable(mflow_pump_name, 160.0_SRK)
+
+  dt=1.0E-1_SRK
+  DO i=1,2000
+    CALL test_fmu2_slave%getNamedVariable(T_out_name, T_out)
+    CALL test_fmu2_slave%getNamedVariable(mflow_out_name, mflow_out)
+    CALL test_fmu2_slave%getNamedVariable(mflow_pump_name, mflow_pump)
+    CALL test_fmu2_slave%getNamedVariable(P_out_name, P_out)
+    CALL test_fmu2_slave%getNamedVariable(T_in_name, T_in_out)
+    CALL test_fmu2_slave%doStep(dt)
+    time=time+dt
+    WRITE(*,*) time, T_in_out, T_out, mflow_out, mflow_pump, P_out
+  ENDDO
+  CALL test_fmu2_slave%setRestart()
+
+  ! Increase the flow rate, keep T_in the same
+  CALL test_fmu2_slave%setNamedVariable(mflow_in_name, 165.0_SRK)
+  CALL test_fmu2_slave%setNamedVariable(mflow_pump_name, 165.0_SRK)
+
+  DO i=1,2000
+    CALL test_fmu2_slave%getNamedVariable(T_out_name, T_out)
+    CALL test_fmu2_slave%getNamedVariable(mflow_out_name, mflow_out)
+    CALL test_fmu2_slave%getNamedVariable(mflow_pump_name, mflow_pump)
+    CALL test_fmu2_slave%getNamedVariable(P_out_name, P_out)
+    CALL test_fmu2_slave%getNamedVariable(T_in_name, T_in_out)
+    CALL test_fmu2_slave%doStep(dt)
+    time=time+dt
+    WRITE(*,*) time, T_in_out, T_out, mflow_out, mflow_pump, P_out
+  ENDDO
+
+  ! call rewind
+  CALL test_fmu2_slave%rewindToRestart()
```

Comment:
The current Transform model is not currently working with setRestartPoint and rewindToRestart.  I can add the suggested checks but they would fail.   The ability to save and rewind the system state is up to the FMU model to implement properly.